### PR TITLE
Add FreeBSD support

### PR DESCRIPTION
**Pull request by LukaszMoskala**
Closes issue #17

### Checklist
- [x] Pull request details were added to CHANGELOG.rst
- [ ] Documentation was updated (if needed)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -13,6 +13,7 @@ version 0.3.0-dev
   improvements and changes with regards to the compression levels. For full
   details checkout the `zlib-ng 2.1.2 release notes
   <https://github.com/zlib-ng/zlib-ng/releases/tag/2.1.2>`_.
++ Added BSD support (tested only with FreeBSD)
```
[the remark this answers is not in the document]
Comment (LukaszMoskala) [reply CHANGELOG.rst:16]: That's fair, give me a moment

**Review by rhpvorderman — APPROVED** [review]

**Comment by rhpvorderman** [comment]
Thanks! I will merge this as soon as the tests pass.

**Comment by gdt** [comment]
Thanks for submitting and merging this!   I tried to update Home Assistant to 2023.12, on NetBSD, and failed.   I have not gotten around to figuring out how to fix.   I just tried again with 2023.12.4, wondering if it had been addressed and was very pleasantly surprised -- the pip upgrade finished straigthforwardly and my instance is running fine.

I also wonder if the Unix (POSIX?) branch should simply be "if not windows and if not mac", rather than an explicit list, but with bsd in the code, it's working for me.

**Comment by rhpvorderman** [comment]
There is also stuff like OS/2, android, iOS. So it has to be a little bit more explicit ;-).

**Pull request merged**